services.sh

```diff
@@ -214,6 +214,99 @@ function manta_setup_common_log_rotation {
+# SERVICE is given, then logs in /var/svc/log for each instance of SERVICE will
+# also be rotated (which are SMF service logs).
+#
+# This works as follows: logadm is configured to rotate the common service logs
+# (i.e., config-agent and registrar) as well as any additional Manta service
+# logs (i.e., whatever's installed in this zone, like buckets-api, mako, or
+# whatever).  The rotated logs are dropped into /var/log/manta/upload.  cron is
+# configured to run logadm every hour on the hour, so each hour's logs get
+# dropped into /var/log/manta/upload at the top of each hour.  Separately, cron
+# is configured to run the log uploader script (backup.sh) every hour, which
+# uploads any logs it finds in /var/log/manta/upload up to Manta and then
+# removes the local copies.  If Manta is down when this happens, the log files
+# will remain in /var/log/manta/upload until the next hour when Manta is up, at
+# which point they'll be uploaded and then the local copies will be removed.
+#
+# This function relies on SERVICE already being set up in order to determine the
+# number of instances of SERVICE so it SHOULD NOT be called until after SERVICE
+# has been initialized.
+function manta_buckets_setup_common_log_rotation {
+    echo "Setting up common log rotation entries"
+
+    #
+    # Create /var/log/manta/upload, where we store files ready to be uploaded.
+    #
+    mkdir -p /var/log/manta/upload
+    chown root:sys /var/log/manta/upload
+
+    #
+    # Copy the log uploader into a place where cron can run it.
+    #
+    local DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+    mkdir -p /opt/smartdc/common/sbin
+    cp ${DIR}/backup.sh /opt/smartdc/common/sbin
+    chmod 755 /opt/smartdc/common/sbin/backup.sh
+    cp ${DIR}/logrotateandupload.sh /opt/smartdc/common/sbin
+    chmod 755 /opt/smartdc/common/sbin/logrotateandupload.sh
+    chown root:sys /opt/smartdc/common
+
+    #
+    # Ensure that log rotation HUPs *r*syslog.
+    #
+    logadm -r /var/adm/messages
+    logadm -w /var/adm/messages -C 4 -a 'kill -HUP `cat /var/run/rsyslogd.pid`'
+
+    #
+    # We want to rotate smf_logs and /var/log/*.log last so they don't override
+    # our log rotation rules for files otherwise caught in their glob pattern,
+    # We'll remove those entries here and re-add them below, in
+    # manta_common_setup_end, after all of our other changes.
+    #
+    # The "/var/log/*.debug" entry is vestigial for old vmadm logs (see
+    # MANTA-3684) and can be removed.
+    #
+    logadm -r smf_logs
+    logadm -r '/var/log/*.log'
+    logadm -r '/var/log/*.debug'
+
+    #
+    # Add the logadm configurations for the config-agent and registrar services
+    # (which are present in every zone), plus the one we've been given.
+    #
+    manta_add_logadm_entry "config-agent"
+    manta_add_logadm_entry "registrar"
+    if [[ $# -ge 1 ]]; then
+        for port in `svcs -H -o fmri $1 | grep -v default | sed s/'.*-'//`
```

Comment:
```suggestion
        for port in `svcs -H -o fmri $1 | grep -v :default | sed s/'.*-'//`
```

for the freak case of there possibly being the string "default" elsewhere in the FMRI.